django/website/logframe/tests/test_api.py

```diff
@@ -41,3 +56,104 @@ def test_editor_can_change_data():
     request = mock.Mock(method="POST", user=u1)
     perm_obj = CanEditOrReadOnly()
     assert perm_obj.has_object_permission(request, None, None) is True
+
+
+def test_id_filter_backend_filter_queryset_filters_on_ids():
+    request = RequestFactory().get('/?id=1&id=2&id=3')
+    request = Request(request)
+    id_filter_backend = IDFilterBackend()
+    mock_queryset = mock.Mock(filter=mock.Mock())
+    id_filter_backend.filter_queryset(request, mock_queryset, None)
+
+    mock_queryset.filter.assert_called_with(id__in=[1, 2, 3])
+
+
+def test_get_period_filter_returns_function():
+    start_date = date.today() - timedelta(days=1)
+    end_date = date.today()
+    ret_val = get_period_filter(start_date, end_date, 'start_date', 'end_date')
+    assert isfunction(ret_val)
+
+
+def test_get_period_filter_function_filters_queryset():
+    start_date = date.today() - timedelta(days=1)
+    end_date = date.today()
+
+    rel_1 = Q(**{'start_date__gte': start_date}) & Q(**{'start_date__lte': end_date})
```

Comment:
Any reason why you used `Q(**{'start_date__gte': start_date})` rather than `Q(start_date__gte=start_date)`? I find the latter more readable, but maybe this is by convention?